driver/docker-container/driver.go

```diff
@@ -205,6 +213,211 @@ func (d *Driver) copyLogs(ctx context.Context, l progress.SubLogger) error {
 	return rc.Close()
 }
 
+// copyToContainer is based on the implementation from docker/cli
+// https://github.com/docker/cli/blob/master/cli/command/container/cp.go
+func (d *Driver) copyToContainer(ctx context.Context, srcPath string, dstPath string) error {
```

Comment:
Could we reuse some of this code in k8s driver? cc @morlay

Reply:
it couldn't.

In k8s, we have to read file data and store it in ConfigMap or Secrets, and mount it into running containter.

I prefix add an interface for Driver to add files, like

```
inteface Driver {
   AddFile(mountPath string, data []byte, secure bool)
}
```
Then diffrent driver to do the mounting

Reply:
Actually, this part might be fine to be driver-specific but the part that modifies the toml file and copies certs under `.docker` should not be specific to the driver but shared.

Reply:
So let driver have ability to add file will be useful.

if the driver not need those files,  it could skip `AddFile`

But driver like `docker-container` and `kubernetes` need, it could be easy to `AddFile` with diffirent ways.

Reply:
At least, `copyBuildKitConfToContainer` is needed to `kubernetes` driver too.

Reply:
> should not be specific to the driver but shared.

Yes, `copyBuildKitConfToContainer` could be an utility func that would be called for any driver that uses `ConfigFile` and returns the handle for `/etc/buildkit` conf. It could then be used with `copyToContainer` for the `container` driver and for `kubernetes` I think a `persistenceVolume` and `persistenceVolumeClaims`.

Reply:
@crazy-max 

`kubernetes` will be use the `configMap`/`secret` with file contents, because the container should be scheduled to any cluster node. 

Anyway. If we want to added this feature to support `container` and `kubernetes` both, consider my proposal.
Then I could help upgrade `kubernetes` driver to support this feature too.

Reply:
I don't think `AddFile()` would work a iiuc `/etc/buildkit` is already mounted in k8s so you can't just pick a path like you  can in container driver.

So shared parts: copy files under `.docker`. Parse toml and create correct config file. Provide list of config + certs to driver

Separate by driver:
- container: copy to container
- kubernetes: fill in configmap https://github.com/docker/buildx/commit/c76b5eac0398a6a186de09318d055fb495a6c192#diff-ad9f131660b45257e1b16c6d5be47cfc1e4bb74f0322fbc6619afedba317288cR119

Reply:
@morlay We need to transform the paths because they are paths in random places of the host system that do not work inside container. I don't understand why you have two volumes in your example when they both modify `/etc/buildkit` . I think we should just have one config mount in `/etc/buildkit` and put everything there.

Reply:
> we can't copy a file in a folder that does not exist in the container with the CopyToContainer Docker API.

Yes, but we could write to temp files before CopyToContainer in `AddFile`

If worry about copy cost. we could design the API like https://docs.min.io/docs/golang-client-api-reference.html#PutObject

```go
type PutFileOptions struct {
   Secure bool
}

type Driver interface {
   PutFile(ctx context.Context, containerAbsPath string, r io.Reader, putOpts PutFileOptions) error
}
```

Reply:
So configmap can not contain subdirectories?

Reply:
@tonistiigi Yes. `/` is not allowed in ConfigMap key. 

For subdirectories, we have to rename the absolute path, and resolve it back with `volumes` and `volumeMounts` like my example did.

This why I prefer have a `PutFile()` interface to make driver could do mounting files,
Then we could just convert and mount files for diffrent drivers in `PutFile()`.